extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -570,11 +617,28 @@ class LifecycleConductor {
     }
 
     isReady() {
-        const state = this._zkClient && this._zkClient.getState();
-        return this._producer && this._producer.isReady() && state &&
-            state.code === zookeeper.State.SYNC_CONNECTED.code &&
-            (!this._kafkaBacklogMetrics || this._kafkaBacklogMetrics.isReady()) &&
-            this._tempCredentialsReady();
+        const zkState = !this.needsZookeeper() ||
+            !!(this._zkClient && this._zkClient.getState());
+
+        const zkConnectState = !this.needsZookeeper() ||
+            (zkState && zkState.code === zookeeper.State.SYNC_CONNECTED.code);
+
+        const components = {
+            zkState,
+            zkConnectState,
+            producer: !!(this._producer && this._producer.isReady()),
+            kafkaBacklogMetrics: (!this._kafkaBacklogMetrics || this._kafkaBacklogMetrics.isReady()),
+            tempCreds: this._tempCredentialsReady(),
+        };
+
+        const allReady = Object.keys(components)
+            .reduce((prev, acc) => prev && components[acc], true);
```

Comment:
```suggestion
        const allReady = Object.values(components).every(v => v);
```

Reply:
Ah, I always forget we're es2017 compliant now.